source/extensions/filters/http/aws_request_signing/aws_request_signing_filter.cc

```diff
@@ -27,22 +31,39 @@ FilterStats Filter::generateStats(const std::string& prefix, Stats::Scope& scope
 }
 
 Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
-  const auto& host_rewrite = config_->hostRewrite();
+  const auto* config = getConfig();
+  const auto& host_rewrite = config->hostRewrite();
   if (!host_rewrite.empty()) {
     headers.setHost(host_rewrite);
   }
 
   try {
-    config_->signer().sign(headers);
-    config_->stats().signing_added_.inc();
+    config->signer().sign(headers);
+    config->stats().signing_added_.inc();
   } catch (const EnvoyException& e) {
     ENVOY_LOG(debug, "signing failed: {}", e.what());
-    config_->stats().signing_failed_.inc();
+    config->stats().signing_failed_.inc();
   }
 
   return Http::FilterHeadersStatus::Continue;
 }
 
+const FilterConfig* Filter::getConfig() const {
+  // Cached config pointer.
+  if (effective_config_) {
+    return effective_config_;
+  }
+
+  effective_config_ = Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfig>(
+      HttpFilterNames::get().AwsRequestSigning, decoder_callbacks_->route());
+  if (effective_config_) {
+    return effective_config_;
+  }
+
+  effective_config_ = config_.get();
+  return effective_config_;
+}
```

Comment:
Since a new filter instance is created for every request, there's no benefit in caching the route specific configurations. So the `effective_config_` can be removed.